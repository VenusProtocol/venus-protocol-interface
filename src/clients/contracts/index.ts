export * from './getters';
export * from './hooks';
