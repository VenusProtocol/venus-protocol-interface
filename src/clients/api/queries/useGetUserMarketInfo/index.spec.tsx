import { waitFor } from '@testing-library/react';
import BigNumber from 'bignumber.js';
import React from 'react';

import fakeAddress from '__mocks__/models/address';
import { assetsInAccount } from '__mocks__/models/assetsInAccount';
import { markets } from '__mocks__/models/markets';
import { vTokenBalanceTreasury } from '__mocks__/models/vTokenBalanceTreasury';
import { vTokenBalancesAccount } from '__mocks__/models/vTokenBalancesAccount';
import { getAssetsInAccount, getMarkets, getMintedVai, useGetVTokenBalancesAll } from 'clients/api';
import renderComponent from 'testUtils/renderComponent';

import useGetUserMarketInfo, { UseGetUserMarketInfoOutput } from '.';

jest.mock('clients/api');

const fakeUserVaiMintedWei = new BigNumber('10000000000000000');

describe('api/queries/useGetUserMarketInfo', () => {
  beforeEach(() => {
    (getMarkets as jest.Mock).mockImplementation(() => ({ markets }));
    (getAssetsInAccount as jest.Mock).mockImplementation(() => assetsInAccount);
    (getMintedVai as jest.Mock).mockImplementation(() => fakeUserVaiMintedWei);

    (useGetVTokenBalancesAll as jest.Mock).mockImplementation(({ account }) => {
      if (account === fakeAddress) {
        return { data: vTokenBalancesAccount };
      }
      return { data: vTokenBalanceTreasury };
    });
  });

  it('calculates totals correctly', async () => {
    let data: UseGetUserMarketInfoOutput['data'] = {
      assets: [],
      userTotalBorrowBalanceCents: new BigNumber(0),
      userTotalBorrowLimitCents: new BigNumber(0),
      userTotalSupplyBalanceCents: new BigNumber(0),
      totalXvsDistributedWei: new BigNumber(0),
      dailyVenusWei: new BigNumber(0),
    };

    const CallMarketContext = () => {
      ({ data } = useGetUserMarketInfo({ accountAddress: fakeAddress }));
      return <div />;
    };

    renderComponent(<CallMarketContext />, {
      authContextValue: { account: { address: fakeAddress } },
    });

    await waitFor(() => expect(data.assets.length > 0).toBe(true));
    expect(data).toMatchSnapshot();
  });
});
