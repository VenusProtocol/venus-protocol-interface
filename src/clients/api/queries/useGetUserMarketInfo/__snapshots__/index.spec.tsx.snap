// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/queries/useGetUserMarketInfo calculates totals correctly 1`] = `
Object {
  "assets": Array [
    Object {
      "borrowApyPercentage": "-4.6398079628683675",
      "borrowBalance": "118.827983172626313436",
      "borrowCapTokens": "0",
      "borrowerCount": 306,
      "collateral": true,
      "collateralFactor": 0.5,
      "exchangeRateVTokens": "48.8639985701266785335",
      "liquidityCents": 199084867,
      "percentOfLimit": "0",
      "reserveFactor": 0.1,
      "reserveTokens": "4082.328070983454972979",
      "supplierCount": 898,
      "supplyApyPercentage": "1.045586251269331278",
      "supplyBalance": "117016.712722872983975881",
      "tokenPriceDollars": "0.67408826",
      "treasuryTotalBorrows": "1008551.9121941992969102",
      "treasuryTotalBorrowsCents": "67985300.361066058614742",
      "treasuryTotalSupply": "19339683254955736",
      "treasuryTotalSupplyCents": "266794650.7810822265519879",
      "vToken": Object {
        "address": "0x74469281310195A04840Daf6EdF576F559a3dE80",
        "asset": "vSxp.svg",
        "decimals": 8,
        "id": "sxp",
        "symbol": "vSXP",
        "underlyingToken": Object {
          "address": "0x75107940Cf1121232C0559c747A986DEfbc69DA9",
          "asset": "sxp.svg",
          "decimals": 18,
          "id": "sxp",
          "symbol": "SXP",
        },
      },
      "walletBalance": "10291786454.9",
      "xvsBorrowApr": "1.051224864211795426",
      "xvsBorrowApy": "1.351224864211795426",
      "xvsPerDay": "49.999999999999968",
      "xvsSupplyApr": "0.14260730848036692",
      "xvsSupplyApy": "0.34260730848036692",
    },
    Object {
      "borrowApyPercentage": "-2.0144969997738267",
      "borrowBalance": "1000.206699",
      "borrowCapTokens": "0",
      "borrowerCount": 232,
      "collateral": true,
      "collateralFactor": 0.81,
      "exchangeRateVTokens": "47.1672416801825796186",
      "liquidityCents": 1.0001830379044496e+21,
      "percentOfLimit": "0",
      "reserveFactor": 0.1,
      "reserveTokens": "47245578337.903163",
      "supplierCount": 638,
      "supplyApyPercentage": "0.000001539982091825",
      "supplyBalance": "0",
      "tokenPriceDollars": "1.00008302",
      "treasuryTotalBorrows": "8586112918853.550179",
      "treasuryTotalBorrowsCents": "858682573794807.340073586058",
      "treasuryTotalSupply": "4.717199913187927152720001083e+28",
      "treasuryTotalSupplyCents": "1.0001838918443319696540393346289641043e+21",
      "vToken": Object {
        "address": "0xD5C4C2e2facBEB59D0216D0595d63FcDc6F9A1a7",
        "asset": "vUsdc.svg",
        "decimals": 8,
        "id": "usdc",
        "symbol": "vUSDC",
        "underlyingToken": Object {
          "address": "0x16227D60f7a0e586C66B005219dfc887D13C9531",
          "asset": "usdc.svg",
          "decimals": 6,
          "id": "usdc",
          "symbol": "USDC",
        },
      },
      "walletBalance": "21328466.322837",
      "xvsBorrowApr": "0.000001125348834693",
      "xvsBorrowApy": "0.000002125348834693",
      "xvsPerDay": "999.9999936",
      "xvsSupplyApr": "8.24635e-13",
      "xvsSupplyApy": "1.824635e-12",
    },
    Object {
      "borrowApyPercentage": "-2.014488268288307",
      "borrowBalance": "27227759.175556",
      "borrowCapTokens": "0",
      "borrowerCount": 315,
      "collateral": true,
      "collateralFactor": 0.8,
      "exchangeRateVTokens": "49.80663521914662316131",
      "liquidityCents": 1.0097990100071567e+26,
      "percentOfLimit": "0",
      "reserveFactor": 0.1,
      "reserveTokens": "5143.61615",
      "supplierCount": 736,
      "supplyApyPercentage": "0",
      "supplyBalance": "0",
      "tokenPriceDollars": "0.9999",
      "treasuryTotalBorrows": "30465941.078326",
      "treasuryTotalBorrowsCents": "3046289448.42181674",
      "treasuryTotalSupply": "5.029972090817266864401527367893625e+33",
      "treasuryTotalSupplyCents": "1.009799010007156842907638340455919692236935e+26",
      "vToken": Object {
        "address": "0xb7526572FFE56AB9D7489838Bf2E18e3323b441A",
        "asset": "vUsdt.svg",
        "decimals": 8,
        "id": "usdt",
        "symbol": "vUSDT",
        "underlyingToken": Object {
          "address": "0xA11c8D9DC9b66E209Ef60F0C8D969D3CD988782c",
          "asset": "usdt.svg",
          "decimals": 6,
          "id": "usdt",
          "symbol": "USDT",
        },
      },
      "walletBalance": "1.0000000000000022287821803419e+22",
      "xvsBorrowApr": "0.200882672824011072",
      "xvsBorrowApy": "0.600882672824011072",
      "xvsPerDay": "999.9999936",
      "xvsSupplyApr": "0",
      "xvsSupplyApy": "0",
    },
    Object {
      "borrowApyPercentage": "-2.8269802937000055",
      "borrowBalance": "13037195.563371302902611447",
      "borrowCapTokens": "0",
      "borrowerCount": 533,
      "collateral": true,
      "collateralFactor": 0.8,
      "exchangeRateVTokens": "49.20026700145764253422",
      "liquidityCents": 970801278578,
      "percentOfLimit": "0",
      "reserveFactor": 0,
      "reserveTokens": "0",
      "supplierCount": 2235,
      "supplyApyPercentage": "0.222026939368167748",
      "supplyBalance": "100002044.854604682831251369",
      "tokenPriceDollars": "1.00019714",
      "treasuryTotalBorrows": "838879295.456576767818085792",
      "treasuryTotalBorrowsCents": "83904467212.0883077362093449",
      "treasuryTotalSupply": "51881576366895517915",
      "treasuryTotalSupplyCents": "1054705745790.4263761515589172",
      "vToken": Object {
        "address": "0x08e0A5575De71037aE36AbfAfb516595fE68e5e4",
        "asset": "vBusd.svg",
        "decimals": 8,
        "id": "busd",
        "symbol": "vBUSD",
        "underlyingToken": Object {
          "address": "0x8301F2213c0eeD49a7E28Ae4c3e91722919B8B47",
          "asset": "busd.svg",
          "decimals": 18,
          "id": "busd",
          "symbol": "BUSD",
        },
      },
      "walletBalance": "9915501406.868175386591055703",
      "xvsBorrowApr": "0.001753286763848276",
      "xvsBorrowApy": "0.021753286763848276",
      "xvsPerDay": "999.9999936",
      "xvsSupplyApr": "0.000730355354365121",
      "xvsSupplyApy": "0.001730355354365121",
    },
    Object {
      "borrowApyPercentage": "-94.54098055818199",
      "borrowBalance": "27338.744865716700054022",
      "borrowCapTokens": "0",
      "borrowerCount": 261,
      "collateral": true,
      "collateralFactor": 0.8,
      "exchangeRateVTokens": "30.21544099330438303214",
      "liquidityCents": 3974,
      "percentOfLimit": "0",
      "reserveFactor": 0.1,
      "reserveTokens": "6133.177004077521280386",
      "supplierCount": 1700,
      "supplyApyPercentage": "86.319696955441989068",
      "supplyBalance": "136076.839108609628017125",
      "tokenPriceDollars": "294.99354795",
      "treasuryTotalBorrows": "163570.361869451489735025",
      "treasuryTotalBorrowsCents": "4644293442.7935753968443612",
      "treasuryTotalSupply": "475703606186365",
      "treasuryTotalSupplyCents": "4644297417.0928637733777829",
      "vToken": Object {
        "address": "0x2E7222e51c0f6e98610A1543Aa3836E092CDe62c",
        "asset": "vBnb.svg",
        "decimals": 8,
        "id": "bnb",
        "symbol": "vBNB",
        "underlyingToken": Object {
          "address": "0x12eb9ef44de5fbea1b2f5ff7a9a375cae9bfb2eb",
          "asset": "bnb.svg",
          "decimals": 18,
          "id": "bnb",
          "isNative": true,
          "symbol": "BNB",
        },
      },
      "walletBalance": "0.37300871",
      "xvsBorrowApr": "1.041107132664179593",
      "xvsBorrowApy": "1.141107132664179593",
      "xvsPerDay": "2999.9999808",
      "xvsSupplyApr": "1.085821799653397884",
      "xvsSupplyApy": "1.185821799653397884",
    },
    Object {
      "borrowApyPercentage": "-19.90282685098207",
      "borrowBalance": "0",
      "borrowCapTokens": "270000",
      "borrowerCount": 77,
      "collateral": true,
      "collateralFactor": 0.6,
      "exchangeRateVTokens": "49.00971793881545425589",
      "liquidityCents": 138400915,
      "percentOfLimit": "0",
      "reserveFactor": 0.25,
      "reserveTokens": "9456.129458228979467263",
      "supplierCount": 326,
      "supplyApyPercentage": "2.509901167951039272",
      "supplyBalance": "0",
      "tokenPriceDollars": "1",
      "treasuryTotalBorrows": "306488.837102836344256144",
      "treasuryTotalBorrowsCents": "30648883.7102836344256144",
      "treasuryTotalSupply": "8238070729630362",
      "treasuryTotalSupplyCents": "168090555.8345572668784761",
      "vToken": Object {
        "address": "0x6d6F697e34145Bb95c54E77482d97cc261Dc237E",
        "asset": "vXvs.svg",
        "decimals": 8,
        "id": "xvs",
        "symbol": "vXVS",
        "underlyingToken": Object {
          "address": "0xB9e0E753630434d7863528cc73CB7AC638a7c8ff",
          "asset": "xvs.svg",
          "decimals": 18,
          "id": "xvs",
          "symbol": "XVS",
        },
      },
      "walletBalance": "7960911.638875779896204111",
      "xvsBorrowApr": "0",
      "xvsBorrowApy": "0",
      "xvsPerDay": "0",
      "xvsSupplyApr": "0",
      "xvsSupplyApy": "0",
    },
    Object {
      "borrowApyPercentage": "-5.7292945208867785",
      "borrowBalance": "50.098633741742514742",
      "borrowCapTokens": "0",
      "borrowerCount": 51,
      "collateral": true,
      "collateralFactor": 0.8,
      "exchangeRateVTokens": "47.13151629968276308865",
      "liquidityCents": 2208005944,
      "percentOfLimit": "0",
      "reserveFactor": 0.1,
      "reserveTokens": "1.766090250815625642",
      "supplierCount": 415,
      "supplyApyPercentage": "1.814867084183460806",
      "supplyBalance": "0",
      "tokenPriceDollars": "30046.80414779",
      "treasuryTotalBorrows": "410.044985006090348771",
      "treasuryTotalBorrowsCents": "1232054135.626148384986397",
      "treasuryTotalSupply": "5387756610941",
      "treasuryTotalSupplyCents": "3434747710.1218290503961459",
      "vToken": Object {
        "address": "0xb6e9322C49FD75a367Fcb17B0Fcd62C5070EbCBe",
        "asset": "vBtcb.svg",
        "decimals": 8,
        "id": "btcb",
        "symbol": "vBTCB",
        "underlyingToken": Object {
          "address": "0xA808e341e8e723DC6BA0Bb5204Bafc2330d7B8e4",
          "asset": "btcb.svg",
          "decimals": 18,
          "id": "btcb",
          "symbol": "BTCB",
        },
      },
      "walletBalance": "77.206518420139934356",
      "xvsBorrowApr": "3.543731091576054636",
      "xvsBorrowApy": "4.543731091576054636",
      "xvsPerDay": "2999.9999808",
      "xvsSupplyApr": "1.306740170918635611",
      "xvsSupplyApy": "1.606740170918635611",
    },
    Object {
      "borrowApyPercentage": "-8.426578436066722",
      "borrowBalance": "21.87357158118604801",
      "borrowCapTokens": "0",
      "borrowerCount": 51,
      "collateral": true,
      "collateralFactor": 0.8,
      "exchangeRateVTokens": "48.0191036019707941683",
      "liquidityCents": 984594834,
      "percentOfLimit": "0",
      "reserveFactor": 0.1,
      "reserveTokens": "27.361500255976792194",
      "supplierCount": 211,
      "supplyApyPercentage": "4.552151837121089047",
      "supplyBalance": "0.000002413831330426",
      "tokenPriceDollars": "2216.20104621",
      "treasuryTotalBorrows": "6946.982350412691554991",
      "treasuryTotalBorrowsCents": "1539590955.2987011849433085",
      "treasuryTotalSupply": "54560058223524",
      "treasuryTotalSupplyCents": "2518082368.1866852001814986",
      "vToken": Object {
        "address": "0x162D005F0Fff510E54958Cfc5CF32A3180A84aab",
        "asset": "vEth.svg",
        "decimals": 8,
        "id": "eth",
        "symbol": "vETH",
        "underlyingToken": Object {
          "address": "0x98f7A83361F7Ac8765CcEBAB1425da6b341958a7",
          "asset": "eth.svg",
          "decimals": 18,
          "id": "eth",
          "symbol": "ETH",
        },
      },
      "walletBalance": "2.027768654221373684",
      "xvsBorrowApr": "0.194444965549789125",
      "xvsBorrowApy": "0.594444965549789125",
      "xvsPerDay": "500.0000256",
      "xvsSupplyApr": "0.163034531797658077",
      "xvsSupplyApy": "0.363034531797658077",
    },
    Object {
      "borrowApyPercentage": "-3.1620166111983328",
      "borrowBalance": "459.14549864903700506",
      "borrowCapTokens": "0",
      "borrowerCount": 15,
      "collateral": true,
      "collateralFactor": 0.6,
      "exchangeRateVTokens": "49.74845950798363258837",
      "liquidityCents": 1002519591,
      "percentOfLimit": "0",
      "reserveFactor": 0.1,
      "reserveTokens": "40.083769609634382585",
      "supplierCount": 150,
      "supplyApyPercentage": "0.314789932511761432",
      "supplyBalance": "96005.08925446772751426",
      "tokenPriceDollars": "77.08032554",
      "treasuryTotalBorrows": "16429.365902695430643694",
      "treasuryTotalBorrowsCents": "126638087.2195539757486425",
      "treasuryTotalSupply": "728570158873914",
      "treasuryTotalSupplyCents": "1128847517.6948646438679408",
      "vToken": Object {
        "address": "0xAfc13BC065ABeE838540823431055D2ea52eBA52",
        "asset": "vLtc.svg",
        "decimals": 8,
        "id": "ltc",
        "symbol": "vLTC",
        "underlyingToken": Object {
          "address": "0x969F147B6b8D81f86175de33206A4FD43dF17913",
          "asset": "ltc.svg",
          "decimals": 18,
          "id": "ltc",
          "symbol": "LTC",
        },
      },
      "walletBalance": "9410",
      "xvsBorrowApr": "6.470759902626424876",
      "xvsBorrowApy": "7.470759902626424876",
      "xvsPerDay": "500.0000256",
      "xvsSupplyApr": "0.511613513165150039",
      "xvsSupplyApy": "0.811613513165150039",
    },
    Object {
      "borrowApyPercentage": "-4.83745439850239",
      "borrowBalance": "0",
      "borrowCapTokens": "0",
      "borrowerCount": 31,
      "collateral": true,
      "collateralFactor": 0.6,
      "exchangeRateVTokens": "48.48115246971846608078",
      "liquidityCents": 703869619,
      "percentOfLimit": "0",
      "reserveFactor": 0.1,
      "reserveTokens": "19878.011501750237063796",
      "supplierCount": 188,
      "supplyApyPercentage": "1.170702197010185572",
      "supplyBalance": "15283991.474622634742699028",
      "tokenPriceDollars": "0.49118518",
      "treasuryTotalBorrows": "5394143.0828219372345264",
      "treasuryTotalBorrowsCents": "264952314.1081648148489552",
      "treasuryTotalSupply": "95528447404746024",
      "treasuryTotalSupplyCents": "967843280.1886978194664069",
      "vToken": Object {
        "address": "0x488aB2826a154da01CC4CC16A8C83d4720D3cA2C",
        "asset": "vXrp.svg",
        "decimals": 8,
        "id": "xrp",
        "symbol": "vXRP",
        "underlyingToken": Object {
          "address": "0x3022A32fdAdB4f02281E8Fab33e0A6811237aab0",
          "asset": "xrp.svg",
          "decimals": 18,
          "id": "xrp",
          "symbol": "XRP",
        },
      },
      "walletBalance": "26757685.698668550913435676",
      "xvsBorrowApr": "3.103840954113091774",
      "xvsBorrowApy": "3.503840954113091774",
      "xvsPerDay": "500.0000256",
      "xvsSupplyApr": "0.547264218064808693",
      "xvsSupplyApy": "0.947264218064808693",
    },
    Object {
      "borrowApyPercentage": "-6.222291162175558",
      "borrowBalance": "477404.693314840191116413",
      "borrowCapTokens": "0",
      "borrowerCount": 19,
      "collateral": true,
      "collateralFactor": 0.6,
      "exchangeRateVTokens": "48.85872192397092956678",
      "liquidityCents": 623332804,
      "percentOfLimit": "0",
      "reserveFactor": 0,
      "reserveTokens": "0",
      "supplierCount": 157,
      "supplyApyPercentage": "2.477129675383148568",
      "supplyBalance": "10223438.720317962598219735",
      "tokenPriceDollars": "0.59959246",
      "treasuryTotalBorrows": "7087754.765387647865700213",
      "treasuryTotalBorrowsCents": "424976431.565550263740894",
      "treasuryTotalSupply": "85423104367735380",
      "treasuryTotalSupplyCents": "1048309232.6563346786922442",
      "vToken": Object {
        "address": "0x37C28DE42bA3d22217995D146FC684B2326Ede64",
        "asset": "vAda.svg",
        "decimals": 8,
        "id": "ada",
        "symbol": "vADA",
        "underlyingToken": Object {
          "address": "0xcD34BC54106bd45A04Ed99EBcC2A6a3e70d7210F",
          "asset": "ada.svg",
          "decimals": 18,
          "id": "ada",
          "symbol": "ADA",
        },
      },
      "walletBalance": "9999990460397.064552020012616879",
      "xvsBorrowApr": "0.354335409204269745",
      "xvsBorrowApy": "0.754335409204269745",
      "xvsPerDay": "174.999999999999744",
      "xvsSupplyApr": "0.105120518892226646",
      "xvsSupplyApy": "0.305120518892226646",
    },
    Object {
      "borrowApyPercentage": "-11.43143544764434",
      "borrowBalance": "2457.30939228",
      "borrowCapTokens": "0",
      "borrowerCount": 20,
      "collateral": true,
      "collateralFactor": 0.8,
      "exchangeRateVTokens": "45.90180050625225121636",
      "liquidityCents": 12654225,
      "percentOfLimit": "0",
      "reserveFactor": 0,
      "reserveTokens": "0",
      "supplierCount": 9,
      "supplyApyPercentage": "10.059455250632296701",
      "supplyBalance": "10892830.35019871",
      "tokenPriceDollars": "0.10127635",
      "treasuryTotalBorrows": "9675013.6298295",
      "treasuryTotalBorrowsCents": "97985006.6629382882325",
      "treasuryTotalSupply": "50145368892336486",
      "treasuryTotalSupplyCents": "110639231.4638646474098287",
      "vToken": Object {
        "address": "0xF912d3001CAf6DC4ADD366A62Cc9115B4303c9A9",
        "asset": "vDoge.svg",
        "decimals": 8,
        "id": "doge",
        "symbol": "vDOGE",
        "underlyingToken": Object {
          "address": "0x67D262CE2b8b846d9B94060BC04DC40a83F0e25B",
          "asset": "doge.svg",
          "decimals": 8,
          "id": "doge",
          "symbol": "DOGE",
        },
      },
      "walletBalance": "1.000000000000009999999999997869620548184657e+34",
      "xvsBorrowApr": "1.120862854912494938",
      "xvsBorrowApy": "1.720862854912494938",
      "xvsPerDay": "91.610000000000064",
      "xvsSupplyApr": "1.122555082433944633",
      "xvsSupplyApy": "1.522555082433944633",
    },
    Object {
      "borrowApyPercentage": "-3.3210326984806846",
      "borrowBalance": "126704.13024174070937539",
      "borrowCapTokens": "0",
      "borrowerCount": 7,
      "collateral": true,
      "collateralFactor": 0.6,
      "exchangeRateVTokens": "49.84526090130182653964",
      "liquidityCents": 109650418,
      "percentOfLimit": "0",
      "reserveFactor": 0.1,
      "reserveTokens": "367.622752501696670187",
      "supplierCount": 152,
      "supplyApyPercentage": "0.250490295580310715",
      "supplyBalance": "1003104.395412477901702707",
      "tokenPriceDollars": "0.80243355",
      "treasuryTotalBorrows": "127045.101864562779374701",
      "treasuryTotalBorrowsCents": "10194525.2099292730251508",
      "treasuryTotalSupply": "7442645752253233",
      "treasuryTotalSupplyCents": "119815375.5117972192153777",
      "vToken": Object {
        "address": "0x3619bdDc61189F33365CC572DF3a68FB3b316516",
        "asset": "vMatic.svg",
        "decimals": 8,
        "id": "matic",
        "symbol": "vMATIC",
        "underlyingToken": Object {
          "address": "0xcfeb0103d4BEfa041EA4c2dACce7B3E83E1aE7E3",
          "asset": "matic.svg",
          "decimals": 18,
          "id": "matic",
          "symbol": "MATIC",
        },
      },
      "walletBalance": "1.000000000009999123122999e+21",
      "xvsBorrowApr": "0",
      "xvsBorrowApy": "0",
      "xvsPerDay": "0",
      "xvsSupplyApr": "0",
      "xvsSupplyApy": "0",
    },
    Object {
      "borrowApyPercentage": "-9.944536924720465",
      "borrowBalance": "1116349.051316481242864469",
      "borrowCapTokens": "50000",
      "borrowerCount": 9,
      "collateral": true,
      "collateralFactor": 0.6,
      "exchangeRateVTokens": "43.23415213861177624648",
      "liquidityCents": 1275027907,
      "percentOfLimit": "0",
      "reserveFactor": 0.25,
      "reserveTokens": "86991.589471047234710721",
      "supplierCount": 195,
      "supplyApyPercentage": "3.62333736102284998",
      "supplyBalance": "1971598.000879882140081876",
      "tokenPriceDollars": "10.61421765",
      "treasuryTotalBorrows": "1116641.764130412123471042",
      "treasuryTotalBorrowsCents": "1185227872.1560157262720313",
      "treasuryTotalSupply": "9645035827191448",
      "treasuryTotalSupplyCents": "2367908342.081922550654256",
      "vToken": Object {
        "address": "0xeDaC03D29ff74b5fDc0CC936F6288312e1459BC6",
        "asset": "vCake.svg",
        "decimals": 8,
        "id": "cake",
        "symbol": "vCAKE",
        "underlyingToken": Object {
          "address": "0xe8bd7cCC165FAEb9b81569B05424771B9A20cbEF",
          "asset": "cake.svg",
          "decimals": 18,
          "id": "cake",
          "symbol": "CAKE",
        },
      },
      "walletBalance": "10099056442.61527168017781011",
      "xvsBorrowApr": "0.06982533135294494",
      "xvsBorrowApy": "0.26982533135294494",
      "xvsPerDay": "174.999999999999744",
      "xvsSupplyApr": "0.034967169585684514",
      "xvsSupplyApy": "0.134967169585684514",
    },
    Object {
      "borrowApyPercentage": "-0.15634638954092597",
      "borrowBalance": "4.95236796334e-7",
      "borrowCapTokens": "0",
      "borrowerCount": 6,
      "collateral": true,
      "collateralFactor": 0.55,
      "exchangeRateVTokens": "4997145456.36480526891693102724",
      "liquidityCents": 419842882,
      "percentOfLimit": "0",
      "reserveFactor": 0,
      "reserveTokens": "0",
      "supplierCount": 144,
      "supplyApyPercentage": "0.001223666324407421",
      "supplyBalance": "0",
      "tokenPriceDollars": "399.24",
      "treasuryTotalBorrows": "83.019490298266506222",
      "treasuryTotalBorrowsCents": "3314470.1306679919944071",
      "treasuryTotalSupply": "5.29651047362260635259e+21",
      "treasuryTotalSupplyCents": "423157352.5296656248989134",
      "vToken": Object {
        "address": "0x714db6c38A17883964B68a07d56cE331501d9eb6",
        "asset": "vAave.svg",
        "decimals": 8,
        "id": "aave",
        "symbol": "vAAVE",
        "underlyingToken": Object {
          "address": "0x4B7268FC7C727B88c5Fc127D41b491BfAe63e144",
          "asset": "aave.svg",
          "decimals": 18,
          "id": "aave",
          "symbol": "AAVE",
        },
      },
      "walletBalance": "0",
      "xvsBorrowApr": "11.754861490074778241",
      "xvsBorrowApy": "14.754861490074778241",
      "xvsPerDay": "25.0000000000000128",
      "xvsSupplyApr": "0.00787839946684478",
      "xvsSupplyApy": "0.10787839946684478",
    },
    Object {
      "borrowApyPercentage": "-0.000407488053549464",
      "borrowBalance": "7000.000289196950794559",
      "borrowCapTokens": "0",
      "borrowerCount": 5,
      "collateral": true,
      "collateralFactor": 0.8,
      "exchangeRateVTokens": "454545417.192151829923123031",
      "liquidityCents": 10130224812,
      "percentOfLimit": "0",
      "reserveFactor": 0.2,
      "reserveTokens": "0.003165772154431927",
      "supplierCount": 138,
      "supplyApyPercentage": "2.6208000003e-8",
      "supplyBalance": "0",
      "tokenPriceDollars": "1",
      "treasuryTotalBorrows": "8279.243179938772394707",
      "treasuryTotalBorrowsCents": "827924.3179938772394707",
      "treasuryTotalSupply": "4.605023592695565994023635e+24",
      "treasuryTotalSupplyCents": "10131052736.4547509365152667",
      "vToken": Object {
        "address": "0x3A00d9B02781f47d033BAd62edc55fBF8D083Fb0",
        "asset": "vTusd.svg",
        "decimals": 8,
        "id": "tusd",
        "symbol": "vTUSD",
        "underlyingToken": Object {
          "address": "0xFeC3A63401Eb9C1476200d7C32c4009Be0154169",
          "asset": "tusd.svg",
          "decimals": 18,
          "id": "tusd",
          "symbol": "TUSD",
        },
      },
      "walletBalance": "10813007000",
      "xvsBorrowApr": "100.564801757396621173",
      "xvsBorrowApy": "200.564801757396621173",
      "xvsPerDay": "50.0000000000000256",
      "xvsSupplyApr": "0.001007365834885316",
      "xvsSupplyApy": "0.009007365834885316",
    },
    Object {
      "borrowApyPercentage": "-2.025978773806436",
      "borrowBalance": "0",
      "borrowCapTokens": "0",
      "borrowerCount": 3,
      "collateral": false,
      "collateralFactor": 0.6,
      "exchangeRateVTokens": "16.66661047629612969482",
      "liquidityCents": 273113474,
      "percentOfLimit": "0",
      "reserveFactor": 0.2,
      "reserveTokens": "1.928237688519600088",
      "supplierCount": 133,
      "supplyApyPercentage": "0.001208241156882074",
      "supplyBalance": "0",
      "tokenPriceDollars": "1",
      "treasuryTotalBorrows": "2058.082101085432106229",
      "treasuryTotalBorrowsCents": "205808.2101085432106229",
      "treasuryTotalSupply": "4555302772101467",
      "treasuryTotalSupplyCents": "273319087.8001371190191733",
      "vToken": Object {
        "address": "0x369Fea97f6fB7510755DCA389088d9E2e2819278",
        "asset": "vTrx.svg",
        "decimals": 8,
        "id": "trx",
        "symbol": "vTRX",
        "underlyingToken": Object {
          "address": "0x19E7215abF8B2716EE807c9f4b83Af0e7f92653F",
          "asset": "trx.svg",
          "decimals": 18,
          "id": "trx",
          "symbol": "TRX",
        },
      },
      "walletBalance": "3.000003203144915042",
      "xvsBorrowApr": "6102.522495213142034502",
      "xvsBorrowApy": "8102.522495213142034502",
      "xvsPerDay": "50.0000000000000256",
      "xvsSupplyApr": "0.134415260865180737",
      "xvsSupplyApy": "0.334415260865180737",
    },
    Object {
      "borrowApyPercentage": "-0.03324775882974874",
      "borrowBalance": "0",
      "borrowCapTokens": "0",
      "borrowerCount": 2,
      "collateral": false,
      "collateralFactor": 0.8,
      "exchangeRateVTokens": "49.84091970453273753229",
      "liquidityCents": 1490100,
      "percentOfLimit": "0",
      "reserveFactor": 0.2,
      "reserveTokens": "0.001485321463606881",
      "supplierCount": 5,
      "supplyApyPercentage": "0.000177293083452232",
      "supplyBalance": "0",
      "tokenPriceDollars": "1",
      "treasuryTotalBorrows": "100.007932372130381683",
      "treasuryTotalBorrowsCents": "10000.7932372130381683",
      "treasuryTotalSupply": "74766395277302",
      "treasuryTotalSupplyCents": "1500100.6345896630306407",
      "vToken": Object {
        "address": "0xF206af85BC2761c4F876d27Bd474681CfB335EfA",
        "asset": "vUst.svg",
        "decimals": 8,
        "id": "ust",
        "symbol": "vUST",
        "underlyingToken": Object {
          "address": "0x5A79efD958432E72211ee73D5DDFa9bc8f248b5F",
          "asset": "ust.svg",
          "decimals": 18,
          "id": "ust",
          "symbol": "UST",
        },
      },
      "walletBalance": "0",
      "xvsBorrowApr": "1.342286892512915770892913642930703690751510720619883895e+37",
      "xvsBorrowApy": "2.342286892512915770892913642930703690751510720619883895e+37",
      "xvsPerDay": "50.0000000000000256",
      "xvsSupplyApr": "53.636157517717037851",
      "xvsSupplyApy": "83.636157517717037851",
    },
    Object {
      "borrowApyPercentage": "-2.014488268288307",
      "borrowBalance": "0",
      "borrowCapTokens": "0",
      "borrowerCount": 0,
      "collateral": false,
      "collateralFactor": 0.55,
      "exchangeRateVTokens": "0.5",
      "liquidityCents": 137500,
      "percentOfLimit": "0",
      "reserveFactor": 0.25,
      "reserveTokens": "0",
      "supplierCount": 3,
      "supplyApyPercentage": "0",
      "supplyBalance": "0",
      "tokenPriceDollars": "1",
      "treasuryTotalBorrows": "0",
      "treasuryTotalBorrowsCents": "0",
      "treasuryTotalSupply": "68750000000",
      "treasuryTotalSupplyCents": "137500",
      "vToken": Object {
        "address": "0x9C3015191d39cF1930F92EB7e7BCbd020bCA286a",
        "asset": "vLuna.svg",
        "decimals": 8,
        "id": "luna",
        "symbol": "vLUNA",
        "underlyingToken": Object {
          "address": "0xf36160EC62E3B191EA375dadfe465E8Fa1F8CabB",
          "asset": "luna.svg",
          "decimals": 6,
          "id": "luna",
          "symbol": "LUNA",
        },
      },
      "walletBalance": "0",
      "xvsBorrowApr": "1.187706392698558375",
      "xvsBorrowApy": "1.587706392698558375",
      "xvsPerDay": "75.0000000000000384",
      "xvsSupplyApr": "16.587706392698558375",
      "xvsSupplyApy": "18.587706392698558375",
    },
  ],
  "dailyVenusWei": "0",
  "totalXvsDistributedWei": "3.110919e+24",
  "userTotalBorrowBalanceCents": "6216414758.467084058156978936843085",
  "userTotalBorrowLimitCents": "13871432693.34798845110528092688405",
  "userTotalSupplyBalanceCents": "18411477984.631676380304982489610572",
}
`;
